sklearn/manifold/t_sne.py

```diff
@@ -413,15 +415,36 @@ def trustworthiness(X, X_embedded, n_neighbors=5, precomputed=False):
     precomputed : bool, optional (default: False)
         Set this flag if X is a precomputed square distance matrix.
 
+        ..deprecated:: 0.20
+            ``precomputed`` has been deprecated in version 0.20 and will be
+            removed in version 0.22. Use ``metric`` instead.
+
+    metric : string, or callable, optional, default 'euclidean'
+        Which metric to use for computing pairwise distances between samples
+        from the original input space. If metric is 'precomputed', X must be a
+        matrix of pairwise distances. Otherwise, see the documentation of
+        argument metric in sklearn.pairwise.pairwise_distances for a list of
+        available metrics. However, using a metric different from 'euclidean'
+        and 'precomputed' seems not standard for this task.
+
     Returns
     -------
     trustworthiness : float
         Trustworthiness of the low-dimensional embedding.
     """
     if precomputed:
+        warnings.warn("The flag 'precomputed' has been deprecated in version"
```

Comment:
Check [scikit-learn contributing guide](http://scikit-learn.org/stable/developers/contributing.html#deprecation) to see howto deprecate the `precomputed` attribute using a decorator.

Reply:
and `precomputed` is not an attribute but a parameter. (my bad. sorry)